web/src/app/schedules/ScheduleCalendarOverrideDialog.js

```diff
@@ -0,0 +1,150 @@
+    name: "Cover Someone's Shifts",
+    helperText: "Have a user take over another's shifts for a specified time.",
+  },
+  temp: {
+    title: 'Create a temporary schedule',
+    desc: 'Replace the entire schedule for a given period of time',
+    name: 'Temporary Schedule',
+    helperText:
+      'Define a fixed shift-by-shift schedule to use for a specified time.',
+  },
+  choose: {
+    title: 'Choose an override action',
+    desc: 'This will create a temporary override to the existing schedule.',
+    name: 'Choose',
+    helperText: 'This will determine which override action you want to take',
+  },
+}
+
+const mutation = gql`
+  mutation ($input: CreateUserOverrideInput!) {
+    createUserOverride(input: $input) {
+      id
+    }
+  }
+`
+
+export default function ScheduleCalendarOverrideDialog(props) {
+  const { variantOptions = ['replace', 'remove', 'add', 'temp'] } = props
+  const [step, setStep] = useState(0)
+  const [value, setValue] = useState({
+    addUserID: '',
+    removeUserID: '',
+    start: DateTime.local().startOf('hour').toISO(),
+    end: DateTime.local().startOf('hour').plus({ hours: 8 }).toISO(),
+    ...props.defaultValue,
+  })
+
+  const [activeVariant, setActiveVariant] = useState(
+    variantOptions[0] || 'replace',
+  )
+
+  const { onNewTempSched } = useContext(ScheduleCalendarContext)
+
+  const notices = useOverrideNotices(props.scheduleID, value)
+
+  const [mutate, { loading, error }] = useMutation(mutation, {
+    variables: {
+      input: {
+        ...value,
+        scheduleID: props.scheduleID,
+      },
+    },
+    onCompleted: props.onClose,
+  })
+
+  const onNext = () => {
+    if (activeVariant === 'temp') {
+      onNewTempSched()
+      props.onClose()
+    } else {
+      setStep(step + 1)
+    }
+  }
+
+  const handleChange = (newValue) => {
+    setValue(newValue)
+  }
+
+  const handleVariantChange = (newVariant) => {
+    setActiveVariant(newVariant)
+  }
+
+  return (
+    <FormDialog
+      onClose={props.onClose}
+      title={
+        step === 0
+          ? variantDetails.choose.title
+          : variantDetails[activeVariant].title
+      }
+      subTitle={
+        step === 0
+          ? variantDetails.choose.desc
+          : variantDetails[activeVariant].desc
+      }
+      errors={nonFieldErrors(error)}
+      notices={notices} // create and edit dialogue
+      loading={loading}
+      form={
+        <ScheduleCalendarOverrideForm
+          scheduleID={props.scheduleID}
+          activeStep={step}
+          value={value}
+          onChange={handleChange}
+          onVariantChange={handleVariantChange}
+          activeVariant={activeVariant}
+          disabled={loading}
+          errors={fieldErrors(error)}
+          variantOptions={variantOptions}
+        />
+      }
+      onSubmit={() => mutate()}
+      onNext={step < 1 ? onNext : null}
```

Comment:
I looked into this a little more. This component uses the onNext+onBack prop to determine if it should show the submit button or a next button (same for the exit/back button). If the onNext prop is null it will show the submit button, if there's a handler it'll show "next". TempSchedDialog.tsx has a good example of this. 

Might be a larger refactor if we want to change the functionality here.